Workspace switching

nmajor25 opened the pull request:
Shortcut Story ID: [sc-33460][sc-33875]

paulfalgout committed:
Remove the organization entity

nmajor25 committed:
Remove workspace filtering from worklist/schedule

nmajor25 commented on test/integration/patients/worklist/worklist.js:
```diff
@@ -2534,126 +2324,6 @@ context('worklist page', function() {
-      .first()
-      .get('.sidebar__label')
-      .should('contain', 'Insurance Plans');
-  });
```
Should these two be removed?? I wasn't 100% sure...

nmajor25 commented on test/integration/patients/worklist/schedule.js:
```diff
@@ -1331,41 +1201,10 @@ context('schedule page', function() {
-      .get('[data-list-region]')
-      .find('.schedule-list__list-row')
-      .should('have.length.above', 0);
-  });
```
Good to remove??

nmajor25 commented on test/integration/patients/worklist/schedule.js:
```diff
@@ -2103,51 +1942,4 @@ context('schedule page', function() {
-      .first()
-      .get('.sidebar__label')
-      .should('contain', 'Insurance Plans');
-  });
```
Good to remove?

Looks like this just tests that the workspace filter is not shown at the top of the list and in the filters sidebar.

paulfalgout commented on src/js/views/patients/schedule/filters_views.js:
```diff
@@ -3,19 +3,15 @@ import { View } from 'marionette';
-
 import './schedule.scss';
 
 const FiltersView = View.extend({
```
Is this view needed?

paulfalgout replied:
Is this file needed even

nmajor25 replied:
I don't think it is. In the last push I removed this `/schedule/filters_views.js` file and the `/worklist/filters_views.js` files. And moved the `AllFiltersButtonView` to the `schedule_views.js` and `worklist_views.js` files.

nmajor25 commented on src/js/views/patients/schedule/schedule.scss:
```diff
@@ -1,17 +0,0 @@
-
-.schedule__list-search {
-  margin-left: auto;
-}
```
I deleted this file and moved these styles over to the `schedule-list.scss` file instead. As it didn't make sense why these were separated.

But I only ended up moving `.schedule__list-search` over. 

The `.schedule__filters` class was removed from the view since it has a wrapper `div` that applies these same exact styles, so it wasn't doing anything.

`.schedule__all-filters:not(:empty)` and `schedule__filter:not(:empty)` were removed since those aren't doing anything anymore with the changes we've made to the views.

nmajor25 commented on src/js/views/patients/worklist/filters_views.js:
```diff
@@ -1,55 +0,0 @@
-  regions: {
-    allFilters: '[data-all-filters-region]',
-    workspace: '[data-workspace-filter-region]',
-    ownerToggle: '[data-owner-toggle-region]',
```
This `ownerToggle` region was moved to the `ListTitleView` a while ago, but wasn't removed from the `FiltersView` here. So this region wasn't be used for anything.

So I removed that region when I moved these views over to the `worklist_views.js` file.

nmajor25 commented on test/integration/patients/worklist/schedule.js:
```diff
@@ -665,29 +638,23 @@ context('schedule page', function() {
     cy
       .get('.list-page__filters')
-      .find('[data-all-filters-region]')
+      .find('.qa-all-filters')
```
If we don't want to use a `.qa-` selector here, I could do this instead:

```js
cy
  .get('[data-filters-region]')
  .find('button')
```

paulfalgout replied:
yep better without it

nmajor25 replied:
In the last push, I removed `.qa-all-filters` and updated the tests to `.find('button')` instead.

paulfalgout committed:
Set minimum heights to nav to keep from overlapping items

paulfalgout committed:
Make root route utility for consistent check

paulfalgout committed:
Make history service with postate and navigate override

Previously we weren’t accurately accounting for browser back/forward.  This does a better, though not perfect job.  (A user can still hold the back button and jump back multiple steps and it’s pretty impossible to know)

paulfalgout committed:
Simplify clinicians collection storage

This is an unrelated refactor

paulfalgout committed:
Add current workspace setting and init to bootstrap

As well as the role request for merge clarity

paulfalgout committed:
Utilize new bootstrap API

paulfalgout committed:
Update backbone.eventrouter

Added a `destroy` for removing registered route handlers and fixed a bug

This needs to be moved to the repo after confidence

paulfalgout committed:
Update router to support workspace slugs as first route arg

paulfalgout committed:
Use workspace header in fetch

paulfalgout committed:
Remove default route handling from patient app

paulfalgout committed:
Add root/workspace route handling to a nav routerapp

paulfalgout committed:
Redirect legacy urls to current workspace

paulfalgout committed:
Add workspace bootstrapping to app frame

And rebuild routes when the current workspace changes.

Also move directories to global

nmajor25 committed:
Rename groups => workspaces in the UI

nmajor25 committed:
Add workspaces to the top nav menu

paulfalgout committed:
Ensure route is in place for start/load for initial and events

There were scenarios where things would get out of order depending on the logic path taken to load a route.

paulfalgout committed:
Update clinicians on workspace switching

paulfalgout committed:
Make owner component load from current clinician

No need to pass workspaces

paulfalgout committed:
Remove unused team logic

paulfalgout committed:
Update worklist state per workspace

We need to store separate for different workspaces because stored state can be invalid between the different workspaces.

Additionally we no longer need default filters so I simplified how state is restored.

paulfalgout committed:
Update default listType to actions

ninja sneak this in.

paulfalgout committed:
Default currentWorkspace settings over org settings

paulfalgout committed:
Filter custom directory filters by setting

The workspace setting would be:
```
custom_filters: [“foo”, “bar”, “baz”]
```
where the value is the directory slug

nmajor25 committed:
Move the dashboards link from the admin tool menu to the left nav

After other dashboard enhancements were made, more users will have access to dashboards. Not only admin users.

paulfalgout committed:
Merge pull request #935 from RoundingWell/owner-component

User current workspace for owner component

paulfalgout committed:
Merge pull request #934 from RoundingWell/top-nav-workspaces

Add workspaces to the top nav menu

nmajor25 committed:
Update app nav min height to accomodate the new dashboards item

paulfalgout committed:
Merge pull request #937 from RoundingWell/dashboard-nav-link

Move dashboards link to the left nav

paulfalgout committed:
Handle worklist droplist onSelect

This was removed, but was still used by the droplist.
I also nested the droplist contents in the scroll class

paulfalgout committed:
Remove changed to the picklist component

Changing the picklist class options broke possibly a poorly named `ui` attribute on the options list.

Perhaps we shouldn’t be using droplist for the main menu since it is so different than the standard

However,
I was able to reuse the `headingText` and `infoText` properties passed to the template to handle the workspace name and help url.
It’s a little dirty, but not crazy dirty and prevents needing to switch from droplist.

This also limits the awkward changes to the custom template.
Fixes [sc-34078]

paulfalgout committed:
Fix roles bootstrap request

At one time `teams:active` was a bootstrap request, and roles was doing a similar filitering so I guess I renamed it, but in the end, it’s only called in one place and isn’t that similar.  Just renamed it back so it isn’t breaking.

Fixes [sc-34082]

paulfalgout committed:
Merge pull request #941 from RoundingWell/bugfix/workspace-switch

Handle worklist droplist onSelect

nmajor25 committed:
Don't trigger event-router event if the user selects the current workspace

paulfalgout committed:
Merge pull request #942 from RoundingWell/select-current-workspace

Don't trigger event-router event if the user selects the current workspace

paulfalgout committed:
Fix history goback

Newest iteration was keeping great track of history, but it wasn’t removing the current route so it would “go back” to itself.

I also added an `handleEmptyHistory` optional handler for if there’s no previous history.  This fixes @woody’s concern with direct links to forms not going back to anywhere useful.

paulfalgout committed:
Remove reduced patient app default route

This shouldn’t have been handling this after the latest changes.  Follows the patient app structure

paulfalgout committed:
Reshow nav droplist on workspace change

Fixes [sc-sc-34124]

Replaces https://github.com/RoundingWell/care-ops-frontend/pull/943

This reshowing needs to within the nav app.  As the nav app is now a router, it’d be best if it didn’t have to restart and rerender everything.  The app should be able to handle the change internally.

paulfalgout committed:
Utilize history handleEmptyHistory for default route

This change looks bigger than it is.  The additional nested function was making the nested success handler in the `listenTo` rather deep, so I used `bindEvents` to make the service handlers more similar to `stateEvents`

This change looks bigger than it is.  The additional nested function was making the nested success handler in the `listenTo` rather deep, so I used `bindEvents` to make the service handlers more similar to `stateEvents`

This change looks bigger than it is.  The additional nested function was making the nested success handler in the `listenTo` rather deep, so I used `bindEvents` to make the service handlers more similar to `stateEvents`

paulfalgout committed:
Merge pull request #946 from RoundingWell/bugfix/render-droplist-workspace-change

Reshow nav droplist on workspace change

paulfalgout committed:
Merge pull request #945 from RoundingWell/bugfix/reduced-app

Remove reduced schedule default handler

paulfalgout committed:
Merge pull request #944 from RoundingWell/bugfix/go-back-history

Fix go back history on forms

nmajor25 committed:
Remove state from the program action & program flow sidebar

paulfalgout committed:
Merge pull request #949 from RoundingWell/remove-program-action-sidebar-state

Remove state status from the program action & program flow sidebar

nmajor25 committed:
When creating a new action, set the initial default state as the first with

Before we just grabbed the first state for this (), which could bite us when states data is in a different order than we expect.

paulfalgout committed:
Merge pull request #950 from RoundingWell/initial-action-state

When a new action is created, set its initial state to the first state with `status: queued`

nmajor25 committed:
Add _state back to program-actions and program-flows entities

nmajor25 committed:
Merge pull request #951 from RoundingWell/program-state

Add `_state:` back to program-actions and program-flows entities

paulfalgout committed:
Set outreach route to root

No workspaces in outreach

paulfalgout committed:
Make bootstrapped fixtures non dynamic

Because these changes entirely with context, it no longer makes sense for them to be dynamic.

paulfalgout committed:
Update stubs to workspace api

paulfalgout committed:
Update Cypress

paulfalgout committed:
Initial test update to remove workspaces as dynamic test content

paulfalgout committed:
Merge pull request #952 from RoundingWell/feature/workspace-switching-test

Feature/workspace switching test
